src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -4279,6 +4284,79 @@ public void testCreateService() throws Exception {
   }
 
   @Test
+  public void testSecretOperations() throws Exception {
+    requireDockerApiVersionAtLeast("1.25", "secret support");
+
+    for (final Secret secret : sut.listSecrets()) {
+      sut.deleteSecret(secret.id());
+    }
+    assertThat(sut.listSecrets().size(), equalTo(0));
+
+    final String secretData = Base64.encodeAsString("testdata".getBytes());
+    
+    final Map<String, String> labels = ImmutableMap.of("foo", "bar", "1", "a");
+
+    final SecretSpec secretSpec = SecretSpec.builder()
+        .name("asecret")
+        .data(secretData)
+        .labels(labels)
+        .build();
+    
+    final SecretCreateResponse response = sut.createSecret(secretSpec);
+    final String secretId = response.id();
+    assertThat(secretId, is(notNullValue()));
+    
+    final SecretSpec secretSpecConflict = SecretSpec.builder()
+        .name("asecret")
+        .data(secretData)
+        .labels(labels)
+        .build();
+
+    try {
+      sut.createSecret(secretSpecConflict);
+      fail("Should fail due to secret name conflict");
+    } catch (ConflictException ex) {
```

Comment:
You’re catching the "correct" exception here according to the spec. But your tests are failing from a docker bug: docker/docker#31909. You'll need to code around it for version 1.13.1 and 17.03.0-ce. 

Adding an additional `catch` clause here for `DockerException` that verifies the docker version is one of these two should be enough.